core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -1006,21 +1235,36 @@ class KafkaZkClient(zooKeeperClient: ZooKeeperClient, isSecure: Boolean) extends
       throw getDataResponse.resultException.get
   }
 
-  private[zk] def createRecursive(path: String): Unit = {
-    val createRequest = CreateRequest(path, null, acls(path), CreateMode.PERSISTENT)
-    var createResponse = retryRequestUntilConnected(createRequest)
-    if (createResponse.resultCode == Code.NONODE) {
+  private[zk] def createRecursive(path: String, data: Array[Byte] = null): CreateResponse = {
```

Comment:
Could we just throw exception instead of returning CreateResponse? None of the caller really needs to check the response.

Reply:
@junrao  previously createRecursive was not throwing any exception if path already exists Some callers may expect exception and some may not.  So I added a flag to control the behavior.

one query:
We are changing AdminUtils interface in this patch. Even though AdminUtils is not a public API, there may be users who may be using AdminUtils in their custom scripts.  Should we add a upgrade note? or we can update the docs during next release.